distributed/shuffle/tests/test_shuffle.py

```diff
@@ -654,10 +657,68 @@ def test_processing_chain():
+            self.value = value
+
+    counter = count()
     workers = ["a", "b", "c"]
     npartitions = 5
-    df = pd.DataFrame({"x": range(100), "y": range(100)})
-    df["_partitions"] = df.x % npartitions
+    df = pd.DataFrame(
+        {
+            f"col{next(counter)}": pd.array([True, False] * 50, dtype="bool"),
+            f"col{next(counter)}": pd.array([True, False] * 50, dtype="boolean"),
+            f"col{next(counter)}": pd.array(range(100), dtype="int8"),
+            f"col{next(counter)}": pd.array(range(100), dtype="int16"),
+            f"col{next(counter)}": pd.array(range(100), dtype="int32"),
+            f"col{next(counter)}": pd.array(range(100), dtype="int64"),
+            f"col{next(counter)}": pd.array(range(100), dtype="uint8"),
+            f"col{next(counter)}": pd.array(range(100), dtype="uint16"),
+            f"col{next(counter)}": pd.array(range(100), dtype="uint32"),
+            f"col{next(counter)}": pd.array(range(100), dtype="uint64"),
+            f"col{next(counter)}": pd.array(range(100), dtype="float16"),
+            f"col{next(counter)}": pd.array(range(100), dtype="float32"),
+            f"col{next(counter)}": pd.array(range(100), dtype="float64"),
+            # f"col{next(counter)}": pd.array(range(100), dtype="csingle"),
+            # f"col{next(counter)}": pd.array(range(100), dtype="cdouble"),
+            # f"col{next(counter)}": pd.array(range(100), dtype="clongdouble"),
+            f"col{next(counter)}": pd.array(
+                [np.datetime64("2022-01-01")] * 100, dtype="datetime64"
+            ),
+            f"col{next(counter)}": pd.array(
+                [np.datetime64("2022-01-01")] * 100,
+                dtype=pd.DatetimeTZDtype(tz="UTC"),
+            ),
+            f"col{next(counter)}": pd.array(
+                [np.timedelta64(1, "D")] * 100, dtype="timedelta64"
+            ),
+            f"col{next(counter)}": pd.array(
+                [pd.Period("2022-01-01", freq="D")] * 100, dtype="period[D]"
+            ),
+            f"col{next(counter)}": pd.array(
+                [pd.Interval(left=0, right=5)] * 100, dtype="Interval"
+            ),
+            f"col{next(counter)}": pd.array(range(100), dtype="Int8"),
+            f"col{next(counter)}": pd.array(range(100), dtype="Int16"),
+            f"col{next(counter)}": pd.array(range(100), dtype="Int32"),
+            f"col{next(counter)}": pd.array(range(100), dtype="Int64"),
+            f"col{next(counter)}": pd.array(range(100), dtype="UInt8"),
+            f"col{next(counter)}": pd.array(range(100), dtype="UInt16"),
+            f"col{next(counter)}": pd.array(range(100), dtype="UInt32"),
+            f"col{next(counter)}": pd.array(range(100), dtype="UInt64"),
+            f"col{next(counter)}": pd.array(["x", "y"] * 50, dtype="category"),
+            # f"col{next(counter)}": pd.array(
+            # [np.nan, np.nan, 1.0, np.nan, np.nan] * 20,
+            # dtype="Sparse[float64]",
+            # ),
```

Comment:
raises `TypeError: Sparse pandas data (column col27) not supported.`

Reply:
https://issues.apache.org/jira/browse/ARROW-8679